packages/editor-ui/src/utils/typeHelpers.ts

```diff
@@ -1,2 +1,5 @@
 export type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
 export type TupleToUnion<T extends readonly unknown[]> = T[number];
+export type HasAtLeastOneKey<T, K extends keyof T = keyof T> = K extends any
+	? Omit<T, K> & Record<K, NonNullable<T[K]>>
+	: never;
```

Comment:
1. `K extends any` is always true, why do we need this condition?
2. Is this working as it should?  The resolution looks like the image below, with both keys optional. I'd expect at least one to be required in each constituent.
3. This is used in a single spot. Maybe we can do away with this helper that needs deciphering and simply list the expected union constituents in the param type. It'd be less abstract and more immediately readable.

<img width="589" alt="Capture 2023-09-27 at 10 41 45@2x" src="https://github.com/n8n-io/n8n/assets/44588767/8b3ee9b3-17ef-4af1-9900-97101e70691d">